src/rsa.rs

```diff
@@ -289,6 +316,31 @@ impl RSAKeyPair {
     pub fn public_modulus_len(&self) -> usize {
         unsafe { RSA_size(self.rsa.as_ref()) }
     }
+
+    /// Pad the message `msg` with the padding algorithm `padding_alg` and write
+    /// the resulting signature into `signature`. The random number generattor `rng`
+    /// is used for blinding (to help make signing constant time).
+    ///
+    /// `signature` must be exactly the length of the `public_modulus_len()`.
+    pub fn sign(&self, padding_alg: &'static RSAPKCS1Padding, rng: &rand::SecureRandom,
+                msg: &[u8], signature: &mut [u8]) -> Result<(), ()> {
+        if signature.len() != self.public_modulus_len() {
+            return Err(());
+        }
+
+        let msg_len = self.public_modulus_len();
+        try!(padding_alg.pad(msg, signature));
+
+        let mut rand = RAND::new(rng);
+        try!(bssl::map_result(unsafe {
+            let blinding = BN_BLINDING_new();
+            let ret = GFp_rsa_private_transform(self.rsa.as_ref(), signature.as_mut_ptr(),
+                                                msg_len, blinding, &mut rand);
```

Comment:
`msg_len` comes from `public_modulus_len()`. Although we verified that that is the same as `signature.len()`, it would still be clearer to use `signature.len()` here. In general, when we pass a slice to a C function, we do so by `x.as_mut_ptr(), x.len()`.